tst/com/amazon/corretto/crypto/provider/test/ServiceSelfTestMetaTest.java

```diff
@@ -88,6 +88,9 @@ public void whenDRBGSelfTestsFail_nothingIsVended() throws Throwable {
         sneakyInvoke(test, "forceFailure");
 
         for (Provider.Service service : accp.getServices()) {
+            if (service.getType().equals("SecureRandom")) {
+                continue;
+            }
```

Comment:
Why do we allow an instance of `SecureRandom` to be advertised when that is the exact thing which failed?

Reply:
I'm not sure I follow? This test looks to me like it uses reflection to test the case where AesCtrDrbg fails it's self tests, and is asserting that all other ACCP services refuse to start up (because they presumably have an implicit dependency on the AesCtrDrbg entropy source?). Let me know if I'm misunderstanding.

However, the new LibCryptoRng is self-sufficient, and doesn't have such a dependency on AesCtrDrbg, and will successfully instantiate itself even if AesCtrDrbg fails. So this test was updated to to not require SecureRandom services fail to instantiate themselves if AesCtrDrbg fails.